docs/contributing/code-contributions.md

```diff
@@ -2,110 +2,102 @@
+## Setting up your local dev environment
 
-To contribute to the blog, check out the setup steps on the [blog contributions](/contributing/blog-contributions/) page. For instructions on contributing to the docs, visit the [docs contributions page](/contributing/docs-contributions/).
+### Install Node and Yarn
 
-## Creating your own plugins and loaders
+- Ensure you have the latest **LTS** version of Node installed (`>= 14.15.0`). `node --version`
+- [Install](https://yarnpkg.com/en/docs/install) the Yarn package manager.
+- Ensure you have the version 1 of Yarn installed (`^1`). `yarn --version`. The Gatsby monorepo wasn't upgraded to Yarn 3 yet.
 
-If you create a loader or plugin, we would love for you to open source it and put it on npm. For more information on creating custom plugins, please see the documentation for [plugins](/docs/plugins/) and the [API specification](/docs/api-specification/).
+### Fork, clone, and branch the repository
 
-## Contributing example sites
+- [Fork](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) the [official `gatsbyjs/gatsby` repository](https://github.com/gatsbyjs/gatsby).
+- Clone your fork: `git clone --depth=1 https://github.com/<your-username>/gatsby.git`
+- Set up repo and install dependencies: `yarn run bootstrap`
+- Create a topic branch: `git checkout -b topics/new-feature-name`
+- Run `yarn run watch` from the root of the repo to watch for changes to packages' source code and compile these changes on-the-fly as you work.
 
-Gatsby's policy is that "Using" example sites (like those in the [examples part of the repo](https://github.com/gatsbyjs/gatsby/tree/master/examples)) should only be around plugins that are maintained by the core team as it's hard to keep things up to date otherwise.
+  - Note that the watch command can be resource intensive. To limit it to the packages you're working on, add a scope flag, like `yarn run watch --scope={gatsby,gatsby-cli}`.
+  - To watch just one package, run `yarn run watch --scope=gatsby`.
+
+### Docs only changes
+
+See [docs setup instructions](/contributing/docs-contributions#docs-site-setup-instructions) for docs-only changes.
+
+### Testing out changes in example project
+
+- Install [gatsby-dev-cli](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-dev-cli):
+
+  - Make sure you have the Gatsby Dev CLI installed with `gatsby-dev -v`
+  - If not, install globally: `yarn global add gatsby-dev-cli`
+  - Set up the `--set-path-to-repo` option to point to your fork
+
+- Run `yarn install` in each of the sites you're testing.
+
+- For each of your Gatsby test sites, run the `gatsby-dev` command inside the test site's directory to copy
+  the built files from your cloned copy of Gatsby. It'll watch for your changes
+  to Gatsby packages and copy them into the site. For more detailed instructions
```

Comment:
Mentioning this briefly like we did above for `yarn watch` before linking to the readme could be useful:

- To copy files from just one package such as `gatsby`, run `gatsby-dev --packages gatsby`